### ISSUE TITLE
Glue results

### ISSUE THREAD

**Issue by honnibal** [explosion/spacy-transformers#31]
We're getting close to being able to put together a complete Glue run. The only task we can't support yet are the regression tasks. It should be a simple change, but I haven't gotten around to it yet. In the meantime, I'll update this thread with a log of experiments. All results refer to the **development set**. I'm running everything on a single Tesla V100.

## `xlnet-cased-base`

* `softmax_class_vector` model architecture
* Learning rate of 0.001 for the output weights
* Learning rate of 2e-5 for the transformer
* Linear learning rate decay
* Variable number of steps per task

### MNLI

I stopped this after 2 epochs as it was taking a long time. It was at around **84.4%**.

### SST-2

Accuracy varies a lot here, but my best checkpoints are up around **94.5%**. Not sure whether the same performance will carry over to the test set, as I'm check-pointing often and cherry-picking the best ones.

### QQP

This is currently training. I set a batch size of 128 and 10 epochs because the dataset is large. ~~Currently at **89.8%**, and still training.~~ Oops, that was the BERT model! Anyway, accuracies on this task aren't very spread out: everyone gets to about 90%, which I think is the noise ceiling. I'll run this last because it's expensive and uninformative.

### RTE

My best runs have gotten to around **72%** on the dev set.

### QNLI

My first run has gotten to **89.8%**.

## `bert-uncased-base`

Unfortunately the BERT base models were trained with the original masking strategy, which did not guarantee that whole words were masked together. This problem makes the masking less effective, because the training objective becomes easier than it should be. It'd be cool to try fine-tuning to correct this.

Currently Devlin has only released whole-word masked models for BERT large :(. This is a shame as IMO the base models are much more useful.

As expected, the `bert-uncased-base` model performs quite a lot worse than the `xlnet-cased-base` model. If a whole-word-masked BERT model is released I think it would be interesting to look at, but until then I think we should focus on the XLNet model.

# Current conclusions

* `xlnet-cased-base` is getting us about the same Glue accuracies as the original BERT large. Nice!
* Setting different learning rates for the output layer and the transformer is definitely recommended.
* Better unfreezing schedules, learning rate schedules etc will probably be quite effective. There seems to be a lot of low-hanging-fruit in the optimization strategy.
* Our scores are pretty good, so the wrapper library can't be doing all that much wrong in terms of the preprocessing?

**Comment by honnibal** [comment]
Trying to put together a run with a single set of hyper-parameters. See here: https://github.com/explosion/spacy-pytorch-transformers/pull/36/ . Current dev scores:

Model                    | CoLa | SST-2 | MRPC | STS-B | QQP | MNLI-m | MNLI-mm | QNLI | RTE | WNLI |
-------------------  | ----- | ------- | ------- | ------ | ----- | ---------- | ----------- | -----  | ----  | ------ |
`xlnet-base-cased` | 60.0 | 94.6     |            |            |         |                |                  |          | 70.8  |
